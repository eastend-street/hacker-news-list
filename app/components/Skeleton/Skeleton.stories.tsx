import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import Skeleton from './Skeleton';

export default {
  title: 'components/Skeleton',
  component: Skeleton,
} as Meta;

const Template: StoryFn<typeof Skeleton> = ({ numOfSkeletons }) => (
  <Skeleton numOfSkeletons={numOfSkeletons} />
);

export const Default = Template.bind({});
Default.args = {
  numOfSkeletons: 10,
};
