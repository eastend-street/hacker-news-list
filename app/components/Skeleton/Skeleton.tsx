export default function Skeleton({
  numOfSkeletons,
}: {
  numOfSkeletons: number;
}) {
  return (
    <div role="status" className="animate-pulse">
      {[...Array(numOfSkeletons)].map((_, index) => (
        <div
          className="flex"
          key={index}
          aria-hidden="true"
          data-testid={`skeleton-node-${index}`}
        >
          <div className="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200" />
          <div className="my-3 h-4 w-96 rounded-full bg-gray-200" />
        </div>
      ))}
      <span className="sr-only">Loading...</span>
    </div>
  );
}
