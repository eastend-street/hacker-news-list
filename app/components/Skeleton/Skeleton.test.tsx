import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as stories from './Skeleton.stories';

const { Default } = composeStories(stories);

describe('components/Skeleton', () => {
  test('should render Skeleton component', () => {
    const { container } = render(<Default />);

    const skeletonElement = screen.getByRole('status');

    expect(skeletonElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('should render Loading... only for screen readers', () => {
    render(<Default />);
    const loadingText = screen.getByText(/Loading.../i);
    expect(loadingText).toBeInTheDocument();
  });

  test('should render 10 skeleton nodes', () => {
    render(<Default />);

    [...Array(10)].forEach((_, index) => {
      const skeletonNode = screen.getByTestId(`skeleton-node-${index}`);
      expect(skeletonNode).toBeInTheDocument();
    });
  });
});
