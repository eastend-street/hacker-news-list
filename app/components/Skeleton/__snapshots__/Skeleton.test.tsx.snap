// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Skeleton should render Skeleton component 1`] = `
<div>
  <div
    class="animate-pulse"
    role="status"
  >
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-0"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-1"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-2"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-3"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-4"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-5"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-6"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-7"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-8"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <div
      aria-hidden="true"
      class="flex"
      data-testid="skeleton-node-9"
    >
      <div
        class="my-3 mx-2 h-4 w-4 rounded-full bg-gray-200"
      />
      <div
        class="my-3 h-4 w-96 rounded-full bg-gray-200"
      />
    </div>
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
